### PR TITLE
Cache session outcome count

### LINKED ISSUE #4066: `ConsensusApi::session_count()` is extremely CPU-intensive
```
$ time RUST_LOG=error ./target/debug/fedimint-cli --data-dir mutinynet-05 dev session-count
{
  "count": 27245
}

real	0m12.442s
```

The federation invite code is `fed11qgqrgvnhwden5te0v9k8q6rp9ekh2arfdeukuet595cr2ttpd3jhq6rzve6zuer9wchxvetyd938gcewvdhk6tcqqysptkuvknc7erjgf4em3zfh90kffqf9srujn6q53d6r056e4apze5cw27h75` if you'd like to verify.

CPU usage spikes to 100% when this call is active.

This causes the `status` RPC, ecash backup, recovery and perhaps other things to be slower than they should since they exercise this.

We should backport a fix for this to `releases/0.2`.

### PR DESCRIPTION
**Pull request by elsirion**
Introduces a DB entry that caches the number of session outcomes in the DB so that we don't have to scan the entire DB every time we need this number.

Fixes #4066

### CODE REVIEW

**Comment by elsirion** [comment]
I ran into some trouble with DB migration tests, could you take a look at 487a5c41fc387c6576e303ae0af99c9979fe8f7f in particular please @m1sterc001guy? Is my attempt to fix it correct or am I missing something?

File: fedimint-core/src/db/mod.rs [hunk 1]
```diff
@@ -1823,6 +1823,8 @@ pub async fn apply_migrations(
 ) -> Result<(), anyhow::Error> {
     let mut dbtx = db.begin_transaction().await;
     let disk_version = dbtx.get_value(&DatabaseVersionKey).await;
+
+    // FIXME: should this be a loop if multiple migrations have to happen?
```
Comment (Maan2003) [inline fedimint-core/src/db/mod.rs:1827]: there is already a loop 

https://github.com/fedimint/fedimint/blob/9c36777c4a0f21742bc99c29c54c12d3f74a4e52/fedimint-core/src/db/mod.rs#L1835

Comment (elsirion) [reply fedimint-core/src/db/mod.rs]: I'm just blind, never mind 🤦‍♂️

**Review by Maan2003 — APPROVED** [review]

**Comment by m1sterc001guy** [comment]
> I ran into some trouble with DB migration tests, could you take a look at https://github.com/fedimint/fedimint/commit/487a5c41fc387c6576e303ae0af99c9979fe8f7f in particular please @m1sterc001guy? Is my attempt to fix it correct or am I missing something?

This is correct, thanks for pointing that out. It hadn't mattered before since we were just reading from old dbs to catch breaking changes, but now with a migration it is needed.

**Review by dpc — APPROVED** [review]
I don't think it's a cache, but LGTM.

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.2`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin releases/v0.2
git worktree add -d .worktree/backport-4072-to-releases/v0.2 origin/releases/v0.2
cd .worktree/backport-4072-to-releases/v0.2
git switch --create backport-4072-to-releases/v0.2
git cherry-pick -x a29d1ff79feeade7c384088b8c1c8a5b5520f421 62b6df4cef5c804d1fd5e61faee510f4d5b25586 edf8d46813780bf8603867a37cb47fb2d905b140
```

**Comment by dpc** [comment]
@fedimint-backports Disappointing.

**Comment by elsirion** [comment]
> I don't think it's a cache

What would you call it? It's redundant/something we could re-build any time, but at considerable cost.

**Comment by douglaz** [comment]
> > I don't think it's a cache
> 
> What would you call it? It's redundant/something we could re-build any time, but at considerable cost.

I call it a "denormalized data"

**Pull request merged**